contracts/stake-cw20/src/contract.rs

```diff
@@ -279,6 +311,40 @@ pub fn execute_fund(
         .add_attribute("amount", amount))
 }
 
+pub fn execute_add_hook(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    addr: String,
+) -> Result<Response, ContractError> {
+    let addr = deps.api.addr_validate(&addr)?;
```

Comment:
Would be SO nice if we could work out a way to validate that the address here is a contract address and not just a wallet address. Unclear to me if CosmWasm supports a way to do this though as none of their code does so far as I can tell.

Reply:
We have that around in some places. Basically, you do a raw query for `ContractInfo`. It's a little tricky to handle the error / success conditions well, though.

Reply:
would be nice, but would rather not have messy validation code and think existing validation is good enough.